app/models/concerns/transient_models_with_purgeable_job_concern.rb

```diff
@@ -0,0 +1,51 @@
+# Archive and Export models are generated in background
+#   those models being are destroy after an expiration period
+#   but, it might take more time to process than the expiration period
+#   this module expose the shared behaviour to compute a job and purge instances
+#   based on a state machine
+module TransientModelsWithPurgeableJobConcern
+  extend ActiveSupport::Concern
+  included do
+    include AASM
+
+    enum job_status: {
+      pending: 'pending',
+      generated: 'generated',
+      failed: 'failed'
+    }
+
+    aasm whiny_persistence: true, column: :job_status, enum: true do
+      state :pending, initial: true
+      state :generated
+      state :failed
+
+      event :make_available do
+        transitions from: :pending, to: :generated
+      end
+      event :restart do
+        transitions from: :failed, to: :pending
+      end
+      event :fail do
+        transitions from: :pending, to: :failed
+      end
+    end
+
+    scope :stale, lambda { |duration|
+      where(job_status: [job_statuses.fetch(:generated), job_statuses.fetch(:failed)])
+        .where('updated_at < ?', (Time.zone.now - duration))
+    }
+
+    def available?
+      generated? && file.attached?
```

Comment:
```suggestion
      generated?
```

Je me demande si on peut faire sauter le `file.attached?` car
- le code qui appelle `make_available` est sensé avoir fait l'attachment `archive_uploader:l16` et `export:compute`
- ca rendrait le concerne utilisable pour des taches périssable sans fichier